src/Contracts/AbstractClient.php

```diff
@@ -92,9 +97,33 @@ protected function fetchFromContainer(string $key): Endpoint
             $this->container[$key] = $endpoint;
         }
 
+        $this->applySslCertificates();
+
         return $this->container[$key];
     }
 
+    /**
+     * Apply SSL certificates when client requires them.
+     */
+    protected function applySslCertificates(): void
```

Comment:
Ik vraag me af of Client de juiste plek is voor deze logica. Dit gaat over (http) communicatie en zou dan ook op die plek geïmplementeerd moeten worden. Of is dit een (aanvullende) vorm van authenticatie?

Reply:
Aanvullend inderdaad, Procura gebruikt een Bearer token en nog eens de certificaten. 
Daarnaast lijkt dit mij juist de juiste plek, dit is het punt net voordat de endpoint classes aangeroepen worden.